### ISSUE TITLE
`message` field ignored for `extensions`

### ISSUE THREAD

**Issue by sir4ur0n** [ndmitchell/hlint#1483]
Hello,

I am trying to get Hlint warnings for the redundant extensions enabled globally (either via `GHC2021` or additional `default-extensions` in Cabal files), as per https://github.com/ndmitchell/hlint/issues/1386

The following works fine:
```yaml
- extensions:
    - default: false
    - name:
        - AllowAmbiguousTypes
```
If I use another extension, e.g. `{-# LANGUAGE OverloadedStrings #-}`, I correctly get an error:
```
$ hlint .
Some/File.hs: Warning: Avoid restricted extensions
Found:
  {-# LANGUAGE OverloadedStrings #-}
Note: may break the code

1 hint
```

I am now trying to customize the `Note:` error message, as per https://github.com/ndmitchell/hlint#restricting-items but to no avail.
E.g. the following:
```yaml
- extensions:
    - default: false
    - message: Hello world
    - name:
        - AllowAmbiguousTypes
```
still prints the exact same error message.

Is my use case possible? If so, how, please? :pray:

**Comment by ndmitchell** [comment]
If you write:

```
- extensions:
  - default: false
  - {name: CPP, within: [], message: "You suck"}
```

That will give you:

```
Sample.hs:1:1-20: Warning: Avoid restricted extensions
Found:
  {-# LANGUAGE CPP #-}
Note: You suck
```

Is that what you are after?

**Comment by sir4ur0n** [comment]
What I am trying to achieve: in `.hlint.yaml`, forbid all extensions by default, and provide a whitelist of allowed extensions. Whenever a forbidden extension is used, I would like to customize the error message (always use the same though, to direct the developers at the `.hlint.yaml` file).

(My original need is actually to report errors whenever an extension is enabled in a file while it's already enabled globally by `GHC2021`, but cf https://github.com/ndmitchell/hlint/issues/1386 this is not yet supported in Hlint)

So given the 2 files:
```yaml
# .hlint.yaml
- extensions:
    - default: false
    - name:
        - AllowAmbiguousTypes
```
```hs
-- Some/File.hs
{-# LANGUAGE OverloadedStrings #-}
```
I expect the following:
```
$ hlint .
Some/File.hs: Warning: Avoid restricted extensions
Found:
  {-# LANGUAGE OverloadedStrings #-}
Note: This is my customized message. If this extension is part of GHC2021, remove it. Otherwise, add it to `.hlint.yaml` to allow it.

1 hint
```
